src/rollup/utils/persist/expiring-local-storage.js

```diff
@@ -0,0 +1,53 @@
+var PAYLOAD_KEY = '$';
+var EXPIRES_KEY = '@';
+
+function safeJsonParse (json) {
+  try {
+    return JSON.parse(json);
+  } catch (e) {
+    // recover
+  }
+}
+
+export default {
+  getItem: function (key) {
+    var value = safeJsonParse(localStorage.getItem(key));
```

Comment:
do we need to ensure `localStorage` exists during init or something and log an error if it doesn't? should be an extreme edge case that we talked about but we still want to prevent errors in the event localStorage doesn't exist (possible i missed the check also)

Reply:
i can feature detect for `localStorage`. During planning, we decided it will be present in the browsers we care about (other than Opera Mini??? but I have never heard of anyone using it) But if `localStorage` is not present, what should we do? Nothing at all? Just exit with an error? That's approximately what will happen if `localStorage` is not present.

Reply:
yah that is kind of what i was thinking "The Pathfora Personalization SDK requires local storage. Unable to initialize." or something along those lines.

Reply:
we are, elsewhere in pathfora, using `sessionStorage` which was introduced alongside `localStorage`... I guess we should have exit early if both are not defined, and claim we depend on the Web Storage API.

Reply:
/me nods

Reply:
Done